src/coreclr/jit/valuenum.cpp

```diff
@@ -7795,6 +7815,18 @@ void Compiler::fgValueNumberTree(GenTree* tree)
 
                         assert(rhsVNPair.GetLiberal() != ValueNumStore::NoVN);
 
+                        ValueNumPair lhsVNPair;
+
+                        if (!lcl->DontVN())
+                        {
+                            lhsVNPair = rhsVNPair;
+                        }
+                        else
+                        {
+                            ValueNum uniqVN = vnStore->VNForExpr(compCurBB, lcl->TypeGet());
+                            lhsVNPair.SetBoth(uniqVN);
+                        }
+
```

Comment:
It appears `lhsVNPair` is unused?

Reply:
interesting, thanks for the catch. I need to investigate why it fixes one of the test cases that I was targeting with this change.